thirdparty/runtime_environments.go

```diff
@@ -204,3 +204,56 @@ func (c *RuntimeEnvironmentsClient) getImageDiff(ctx context.Context, opts Image
 	}
 	return filteredChanges, nil
 }
+
+// Toolchain represents a toolchain's information.
+type Toolchain struct {
+	Name    string
+	Version string
+	Manager string
+}
+
+// ToolchainFilterOptions represents the filtering arguments, each of which is optional except for the AMI.
+type ToolchainFilterOptions struct {
+	AMI     string
+	Page    int
+	Limit   int
+	Name    string // Filter by the name of the toolchain (ex. golang).
+	Version string // Filter by the version (ex. go1.8.7).
+}
+
+// getToolchains returns a list of toolchains from the AMI and filters in the ToolchainFilterOptions.
+func (c *RuntimeEnvironmentsClient) getToolchains(ctx context.Context, opts ToolchainFilterOptions) ([]Toolchain, error) {
+	if opts.AMI == "" {
+		return nil, errors.New("no AMI provided")
+	}
+	params := url.Values{}
+	params.Set("ami", opts.AMI)
+	params.Set("page", strconv.Itoa(opts.Page))
+	if opts.Limit != 0 {
+		params.Set("limit", strconv.Itoa(opts.Limit))
+	}
+	params.Set("name", opts.Name)
+	params.Set("version", opts.Version)
+	params.Set("type", "Toolchains")
```

Comment:
I wonder if we want to reuse the variable (at the top) here? Though we should probably rename the variables since they're not strictly associated with a change